wp/plugin/wp.ml

```diff
@@ -152,8 +153,11 @@ let analyze_proj (ctx : Z3.context) (var_gen : Env.var_gen) (proj : project)
   let pre, env = Pre.visit_sub env post main_sub in
   let pre = Constr.mk_clause [Z3_utils.mk_smtlib2_single env flags.pre_cond] [pre] in
   let pre = Constr.mk_clause hyps [pre] in
-  Format.printf "\nSub:\n%s\nPre:\n%a\n%!"
-    (Sub.to_string main_sub) Constr.pp_constr pre;
+  (* Print statement for constrain-style prover output: *)
```

Comment:
I think we can move this to `Pre.check` so we can handle the print_constr flag in one place by iterating over the list there. The constraint should be the same in both locations.

Reply:
Bonus of being able to print out the `Constr.t` type for both single and comparative analysis as well.

Reply:
The print call on line 160 in wp.ml depends on the value `main_sub`, which is defined in analyze_proj. I can change analyze_proj to output `main_sub` so that it can be called in Pre.check if you feel that isn't too messy.

Reply:
What if we move the printing of `pre` into `Pre.check` and leave the printing of `main_sub` here? It could mirror how `compare_projs` also prints the subroutines. https://github.com/draperlaboratory/cbat_tools/blob/6f6db38da139d5514b9b8ad7e2765a273ef6b6c9/wp/plugin/wp.ml#L258

Reply:
Great idea! I have moved the printing of `pre` into `Pre.check`, and in doing so, when `Pre.check` is called in `wp.ml`, it prints either the pre from `compare_projs` or `analyze_proj`, allowing us to print `pre` for comparative analysis.

Reply:
Looks super good! Do you mind removing the print for `pre` from `wp.ml` so we don't have this code in two places? It looks like we are printing out `pre` twice when I set the flag to `internal`. I think we can leave a simple `printf` for the `main_sub` here.

Reply:
I didn't catch that. 
I'm on it.